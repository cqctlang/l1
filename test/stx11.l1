@define domatch(e)
{
	@local id;

	id = 'foo;

	switch(e){
	@match #id('bar): 
		printf("matched id bar\n");
		break;
	@match #id(x) && x == id:
		printf("matched id foo\n");
		break;
	@match #id(x):
		printf("matched some other id (%a)\n", x);
		break;
	@match #val('bar):
	        printf("match val bar\n");
		break;	
	@match #val(x) && x == id:
	        printf("match val foo\n");
		break;	
	@match #val(x):
		printf("matched some other val (%a)\n", x);
		break;
	default:
		printf("no match\n");
		break;
	}
}


domatch(#id('bar));
domatch(#id('foo));
domatch(#id('other));

domatch(#val('bar));
domatch(#val('foo));
domatch(#val('other));

