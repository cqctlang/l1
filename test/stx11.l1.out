matched id bar
matched id foo
matched some other id (other)
match val bar
match val foo
matched some other val (other)
